src/mirror.c

```diff
@@ -74,13 +74,10 @@ static int write_to_path(char *content, char *path)
 	dir = dirname(tmp);
 	/* attempt to  make the directory
 	 * ignore EEXIST errors */
-	int ret = mkdir(dir, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
-	char *cmd;
-	string_or_die(&cmd, "mkdir -p %s", dir);
-	ret = system(cmd);
-	free_string(&cmd);
+	int ret = mkdir(dir, S_IRWXU | S_IRWXG | S_IRWXO);
 	free_string(&tmp);
-	if (ret) {
+
+	if (ret && EEXIST != errno) {
```

Comment:
May need to expand on this. Line 128 in the error path (unset_mirror_url()) means that a pre-existing _file_ /etc/swupd would be unlinked, although maybe that's not necessarily fatal.